### DISCUSSION TITLE
I hope Mono will have the ability to compile Csharp into native code in the future

### DISCUSSION THREAD

**Discussion by bingbingmasdhg** [mono/mono#21710] [Ideas]
I hope Mono will have the ability to compile CSharp into native code in the future
Hello, I really like this open source tool, but I see that Microsoft has already implemented the function of compiling CSharp into native code in Net7.0. I hope you can also have this function in the future because I don't want to confuse it, it is very troublesome and prone to errors. I hope others cannot extract my script. Just like CPP

**Comment by jbatt33** [comment]
Do you mean you want a way to deploy code without the mono runtime?

**Comment by bingbingmasdhg** [comment]
Yes, its route is as follows:
C # -->assembly code -->local machine code (without any intermediate code, intermediate code is very dangerous, such as Java.)
Of course, there are also drawbacks, such as the inability to use reflection and thermal updates, as well as some convenient functions.
But I won't worry about these, I can give them up for safety reasons.
Microsoft's net AOT has already implemented a similar approach. Although only Windows and Linux are supported in net7.0, with the birth of 8.0, mobile platforms also support it. This is a great thing, and I believe this is the future path of C #.
Because we attach great importance to the confidentiality of our code, in my country, some people may steal the other party's source code and apply for copyright protection, and then sue the original author. This has been a malicious incident that has occurred many times, and we very much hope to avoid it.
Perhaps someone asks, why don't you confuse? For small projects, confusion is easy, but as the project grows larger, it becomes increasingly difficult. We don't want to add a huge workload.
And confusing the code does not prevent the other party from understanding my program framework, they can still simulate my functional implementation.
Perhaps some people ask why C++is not used because it is not easy to use for game logic. It is too complex. If our engine is to be released to other users for use, such as collaborating on game development, it must simplify the script to make it easy to use.
C # is a very suitable choice. If the official can implement this, I would be happy to embed it into my engine.
thanks
